contracts/BaseJumpRateModelV2.sol

```diff
@@ -162,21 +191,22 @@ abstract contract BaseJumpRateModelV2 is InterestRateModel {
         uint256 jumpMultiplierPerYear,
         uint256 kink_
     ) internal {
-        baseRatePerBlock = baseRatePerYear / BLOCKS_PER_YEAR;
-        multiplierPerBlock = multiplierPerYear / BLOCKS_PER_YEAR;
-        jumpMultiplierPerBlock = jumpMultiplierPerYear / BLOCKS_PER_YEAR;
+        uint256 slotsInOneYear = getSlotsInOneYear();
```

Comment:
A public view might be useful for those who want to compute interest rate per year though 🤔

Should we set blocksPerYear immutable in TimeManager to SECONDS_PER_YEAR for time based contracts in this case? So that people have access to this info somehow.

Reply:
Makes sense. And make blocksPerYear private to avoid confusion?

Reply:
`
      // SPDX-License-Identifier: BSD-3-Clause
      pragma solidity 0.8.13;

     import { SECONDS_PER_YEAR } from "./lib/constants.sol";

     abstract contract TimeManager {
    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
    uint256 public immutable slotsInOneYear;

    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
    bool public immutable isTimeBased;

    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
    function() view returns (uint256) internal immutable _getCurrentSlot;

    /// @notice Thrown on invaid arguments
    error InvalidBlocksPerYear();

    /**
     * @param timeBased_ A boolean indicating whether the contract is based on time or block.
     * @param blocksPerYear_ The number of blocks per year
     * @custom:error InvalidBlocksPerYear is thrown if blocksPerYear entered is zero
     * @custom:oz-upgrades-unsafe-allow constructor
     */
    constructor(bool timeBased_, uint256 blocksPerYear_) {
        if (blocksPerYear_ == 0) {
            revert InvalidBlocksPerYear();
        }

        isTimeBased = timeBased_;
        slotsInOneYear = timeBased_ ? SECONDS_PER_YEAR : blocksPerYear_;
        _getCurrentSlot = timeBased_ ? _getBlockTimestamp : _getBlockNumber;
    }

    /**
     * @notice Returns the current timestamp in seconds
     * @return The current timestamp
     */
    function _getBlockTimestamp() private view returns (uint256) {
        return block.timestamp;
    }

    /**
     * @notice Returns the current block number
     * @return The current block number
     */
    function _getBlockNumber() private view returns (uint256) {
        return block.number;
    }

    /**
     * @dev Function to simply retrieve block number or block timestamp
     *  This exists mainly for inheriting test contracts to stub this result.
     * @return Current block number or block timestamp
     */
    function getBlockNumberOrTimestamp() public view virtual returns (uint256) {
        return _getCurrentSlot();
    }
}
`

In my opinion this shoud  be the most optimized verison of **TimeManager**. What are your thoughts?
@chechu @kkirka

Reply:
I think we should only check for `if (blocksPerYear_ == 0)` as after this check, value of blocksOrSecondsPerYear is updated to either `SECONDS_PER_YEAR` or `blockPerYear` passed as an argument.
https://github.com/VenusProtocol/isolated-pools/commit/6b98b095e5e6b3322b6d0c825ba273612e49948a

Reply:
Re: `if (!timeBased_ && blocksPerYear_ == 0)`

I think it depends on what pattern we expect the contract users to follow for `timeBased_` versions. Currently, if it's `timeBased_`, we ignore `blocksPerYear_` but only if it is nonzero, otherwise  if it is zero we revert. This might indeed be confusing. I see two solutions here to avoid confusion:

1. We could explicitly say that "if `timeBased_ == true`, `blocksPerYear_` parameter is ignored" in natspec, and indeed allow any value (including zero) in this case.
2. We could require blocksPerYear_ to be some specific value (either SECONDS_PER_YEAR or zero or max uint or whatever). In this case, the doc would say "if `timeBased_ == true`, `blocksPerYear_` MUST be SECONDS_PER_YEAR" (or "zero", or whatever).